src/spikeinterface/core/core_tools.py

```diff
@@ -934,3 +941,72 @@ def convert_bytes_to_str(byte_value: int) -> str:
         byte_value /= 1024
         i += 1
     return f"{byte_value:.2f} {suffixes[i]}"
+
+
+def spike_vector_to_dict(spike_vector: np.ndarray) -> list[dict]:
+    """
+    Computes all spike trains for all units/segments from a spike vector.
+
+    Parameters
+    ----------
+    spike_vector: np.ndarray
+        The spike vector to convert.
+
+    Returns
+    -------
+    spike_trains: list[dict]:
+        A list containing, for each segment, the spike trains of all units.
+    """
+    assert HAVE_NUMBA, "spike_vector_to_dict() requires `numba`!"
+
+    spike_trains = _vector_to_dict(
+        spike_vector["sample_index"].astype(np.int64),
+        spike_vector["unit_index"].astype(np.int64),
+        spike_vector["segment_index"].astype(np.int64),
+    )
+
+    return spike_trains
+
+
+if HAVE_NUMBA:
+
+    @numba.jit((numba.int64[::1], numba.int64[::1], numba.int64[::1]), nopython=True, nogil=True, cache=True)
+    def _vector_to_dict(sample_index, unit_index, segment_index):
+        """
+        Fast method to convert a spike vector into spike train for all segments and units.
+
+        Parameters
+        ----------
+        sample_index: array
+            spike_vector["sample_index"]
+        unit_index: array
+            spike_vector["unit_index"]
+        segment_index: array
+            spike_vector["segment_index"]
+
+        Returns
+        -------
+        spike_trains: list[dict]
+            A list containing, for each segment, the spike trains of all units.
+        """
+
+        spike_trains = []
+        n_units = 1 + np.max(unit_index)
+
+        for seg in range(1 + segment_index[-1]):
+            n_spikes = np.zeros(n_units, dtype=np.int32)
+            for i in range(len(sample_index)):
+                n_spikes[unit_index[i]] += 1
+
+            spk_trains = numba.typed.Dict()
+            for i in range(n_units):
+                spk_trains[i] = np.empty(n_spikes[i], dtype=np.int64)
+
+            ind = np.zeros(n_units, dtype=np.int64)
+            for i in range(len(sample_index)):
```

Comment:
Can you use enumerate and some better variable name that `ind`. I think something like `current_x` or `counter_x` where x is the purpose would make your intent more transparent and easier to read.

you have indexes, ind, i, all over the place so it is hard to know what is doing what.

Reply:
Done :+1:

Reply:
This is a suggestion btw. If you feel strongly about having a variable `ind`, that's up to you.

Reply:
I feel strongly that I do not know how to name my variables :joy: 
So having someone to double check whether it's undrstandable is a good idea ahah

Reply:
Thanks!

Reply:
The heuristic is to be specific. What is this representing or used for?

There is no gain for conciseness in programming.

Reply:
> The heuristic is to be specific. What is this representing or used for?
> There is no gain for conciseness in programming.

I half-agree. Sometimes to be specific the variable name takes 30 characters which I find too much and makes it un-readable.
But I agreed with you with the variables you mentioned above :)